automation/test-linux.sh

```diff
@@ -0,0 +1,182 @@
+      - ReadWriteOnce
+    resources:
+      requests:
+        storage: 30Gi
+EOF
+
+timeout=600
+sample=10
+current_time=0
+while [ $(oc get pods -n $namespace | grep "${TARGET}-datavolume-original.*Running" | wc -l ) -eq 0 ] ; do 
+  oc get pods -n $namespace
+  current_time=$((current_time + sample))
+  if [ $current_time -gt $timeout ]; then
+    error=true
+    break
+  fi
+  sleep $sample;
+done
+
+current_time=0
+while [ $(oc get pods -n $namespace | grep "${TARGET}-datavolume-original.*Running" | wc -l ) -eq 1 ] ; do 
+  oc get pods -n $namespace
+  current_time=$((current_time + sample))
+  if [ $current_time -gt $timeout ]; then
+    error=true
+    break
+  fi
+  sleep $sample;
+done
+
+sizes=("tiny" "small" "medium" "large")
+workloads=("desktop" "server" "highperformance")
+
+if [[ $TARGET =~ rhel6.* ]]; then
+  workloads=("desktop" "server")
+fi
+
+if [[ $TARGET =~ centos6.* ]]; then
+  workloads=("server")
+fi
+
+if [[ $TARGET =~ ubuntu.* ]]; then
+  workloads=("desktop")
+fi
+
+if [[ $TARGET =~ opensuse.* ]]; then
+  workloads=("server")
+fi
+
+if [[ $TARGET =~ centos7.* ]] || [[ $TARGET =~ centos8.* ]]; then
+  workloads=("server" "desktop")
+fi
+
+delete_vm(){
+  vm_name=$1
+  local template_name=$2
+  set +e
+  #stop vm
+  ./virtctl stop $vm_name -n $namespace
+  #delete vm
+  oc process -n $namespace -o json $template_name NAME=$vm_name SRC_PVC_NAME=$TARGET-datavolume-original SRC_PVC_NAMESPACE=kubevirt | \
+    oc delete -n $namespace -f -
+  set -e
+  #wait until vm is deleted
+  while oc get -n $namespace vmi $vm_name 2> >(grep "not found") ; do sleep $sample; done
+}
+
+run_vm(){
+  vm_name=$1
+  template_path="dist/templates/$vm_name.yaml"
+  local template_name=$( oc get -n ${namespace} -f ${template_path} -o=custom-columns=NAME:.metadata.name --no-headers -n kubevirt )
+  running=false
+
+  #If first try fails, it tries 2 more time to run it, before it fails whole test
+  for i in `seq 1 3`; do
+    error=false
+    oc process -n $namespace -o json $template_name NAME=$vm_name SRC_PVC_NAME=$TARGET-datavolume-original SRC_PVC_NAMESPACE=kubevirt | \
+    jq 'del(.items[0].spec.dataVolumeTemplates[0].spec.pvc.accessModes) |
+    .items[0].spec.dataVolumeTemplates[0].spec.pvc+= {"accessModes": ["ReadWriteOnce"]} | 
+    .items[0].metadata.labels["vm.kubevirt.io/template.namespace"]="kubevirt"' | \
+    oc apply -n $namespace -f -
+
+    # start vm
+    ./virtctl start $vm_name -n $namespace
+    # wait until import pod is running
+    current_time=0
+    while [ $(oc get pods -n $namespace | grep "cdi-upload-$vm_name.*Running" | wc -l ) -eq 0 ] ; do 
+      oc get pods -n $namespace
+      current_time=$((current_time + sample))
+      if [ $current_time -gt $timeout ]; then
+        error=true
+        break
+      fi
+      sleep $sample;
+    done
+
+    # wait until import is finished
+    current_time=0
+    while [ $(oc get pods -n $namespace | grep "cdi-upload-$vm_name.*Running" | wc -l ) -gt 0 ] ; do 
```

Comment:
These loops don't need to run, if the `error` variable is true. It would just add more time to the test for no reason.